File: agentguard/schemas/context.py
"""Runtime context carried alongside every intercepted event."""

from __future__ import annotations

import uuid
from typing import Any

from pydantic import BaseModel, Field


class RuntimeContext(BaseModel):
    """Identity, policy and scope information for the current agent run.

    A single context object is created when a :class:`~agentguard.AgentGuard`
    session starts and is threaded through the event bus, middleware, PEP and
    audit subsystems.
    """

    session_id: str = Field(default_factory=lambda: uuid.uuid4().hex)
    user_id: str | None = None
    agent_id: str | None = None

    policy: str = "default"
    goal: str | None = None
    scope: list[str] = Field(default_factory=list)

    sandboxed: bool = True
    fail_open: bool = True

    tags: list[str] = Field(default_factory=list)
    metadata: dict[str, Any] = Field(default_factory=dict)

    def child(self, **overrides: Any) -> "RuntimeContext":
        """Derive a sub-context (e.g. for a spawned sub-agent or skill)."""
        data = self.model_dump()
        data.update(overrides)
        return RuntimeContext(**data)

File: src/client/js/agentguard/schemas/context.js
"use strict";

class RuntimeContext {
  constructor(data = {}) {
    this.session_id = data.session_id || data.sessionId || "unknown";
    this.user_id = data.user_id ?? data.userId ?? null;
    this.agent_id = data.agent_id ?? data.agentId ?? null;
    this.task_id = data.task_id ?? data.taskId ?? null;
    this.policy = data.policy ?? null;
    this.policy_version = data.policy_version ?? data.policyVersion ?? null;
    this.environment = data.environment ?? null;
    this.metadata = { ...(data.metadata || {}) };
  }

  toDict() {
    return {
      session_id: this.session_id,
      user_id: this.user_id,
      agent_id: this.agent_id,
      task_id: this.task_id,
      policy: this.policy,
      policy_version: this.policy_version,
      environment: this.environment,
      metadata: { ...this.metadata },
    };
  }

  child(overrides = {}) {
    return new RuntimeContext({ ...this.toDict(), ...overrides });
  }

  static fromDict(data = {}) {
    return new RuntimeContext(data);
  }
}

module.exports = {
  RuntimeContext,
};
